dev-python/nbdime/nbdime-3.1.1-r2.ebuild

```diff
@@ -0,0 +1,108 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{9..11} )
+
+inherit distutils-r1 pypi
+
+DESCRIPTION="Diff and merge of Jupyter Notebooks"
+HOMEPAGE="https://jupyter.org https://github.com/jupyter/nbdime"
+SRC_URI+="
+	https://github.com/jupyter/nbdime/commit/b3639a272a156fc1bd3e963e28af2c8ef3558b2e.patch
+		-> ${P}-jupyter_server2-1.patch
+	https://github.com/jupyter/nbdime/commit/51feb2ece5437b2831ffe6282bdce8930de136cf.patch
+		-> ${P}-jupyter_server2-2.patch
+	https://github.com/jupyter/nbdime/commit/c4e1ddd0125a4358b19c980ebfe896a081446b8a.patch
+		-> ${P}-jupyter_server2-3.patch
+	https://github.com/jupyter/nbdime/commit/2f4bf944058048c18202edccb9f43ac4ec1de4fb.patch
+		-> ${P}-jupyter_server2-4.patch
+	https://github.com/jupyter/nbdime/commit/369bce46f2784a345900e42a656d5a9d05ef7e41.patch
+		-> ${P}-jupyter_server2-5.patch
+	https://github.com/jupyter/nbdime/commit/73b70c3bf32cf3dbac9b6660ad61dfd9fc7c64eb.patch
+		-> ${P}-jupyter_server2-6.patch
+	https://github.com/jupyter/nbdime/commit/5c5fda9547eb84c1a38acd2446b99e301b8cda29.patch
+		-> ${P}-jupyter_server2-7.patch
```

Comment:
Any reason you can't use the whole PR, i.e. `https://github.com/jupyter/nbdime/pull/NNN.patch`?

Reply:
Two reasons:

1. The irrelevant content I mentioned in https://github.com/gentoo/gentoo/pull/30010#issuecomment-1461453821, I don't think the modification of ts has anything to do with the bug we are having

2. https://github.com/jupyter/nbdime/pull/641.patch failed to patch

```
patching file nbdime/webapp/nb_server_extension.py
Hunk #1 succeeded at 10 with fuzz 2.
Hunk #2 FAILED at 151.
1 out of 2 hunks FAILED -- saving rejects to file nbdime/webapp/nb_server_extension.py.rej
```

Reply:
Hmm, ok, here's another idea: let's take a fresh snapshot instead, i.e. something like `nbdime-3.1.1_p20230115.ebuild`:

```
EGIT_COMMIT="2da614b603b694e13fc8d597eef1cc09ff07cb10"
MY_P="${PN}-${EGIT_COMMIT}"
# ...
SRC_URI="
	https://github.com/jupyter/nbdime/archive/${EGIT_COMMIT}.tar.gz
		-> ${MY_P}.gh.tar.gz
"
S=${WORKDIR}/${MY_P}
```

Reply:
This sounds good, We can take a snapshot before the upstream tag.

Reply:
I see another error with github snapshot

```
FAILED nbdime/tests/test_js_artifacts_installed.py::test_nbdime_js_artifacts_present - ValueError: missing files: ['/var/tmp/portage/dev-python/nbdime-3.1.1_p20230115/work/nbdime-2da614b603b694e13fc8d597eef1cc09ff07cb10/nbdime/webapp/static/nbdime.js']
```

`nbdime/webapp/static/nbdime.js` is generated by `npm install; npm run build`,  and only provided by pypi source tarball.
Only `dev-python/nbval` depends on nbdime, nbval-0.10.0-r1 can also pass the test after ignore `nbdime/tests/test_js_artifacts_installed.py::test_nbdime_js_artifacts_present`.
So I think we can ignore this test too.

Reply:
Ah, no, that's bad idea then. We don't want to miss installed files.

Reply:
Hmm, looks like the only way we can do it is to patch on dev-python/nbdime/nbdime-3.1.1-r2.ebuild...

Reply:
What do you think of https://github.com/gentoo/gentoo/commit/aad3d643eb4431ac57ffb92c699ac551d4505ef1, is there anything else that needs to be changed? i will push it back.

Reply:
One more idea: would it be a problem if you forked the nbdime github repo and cherry-picked + combined the commits on top of the upstream tag? Then you could use a single `.patch` in `SRC_URI`.

Reply:
Just a simple merge of the commits, I don't think there will be a problem.